test/server/cards/02_SHD/leaders/BoKatanKryzePrincessInExile.spec.js

```diff
@@ -0,0 +1,154 @@
+                this.player1.clickCard(this.bokatanKryze);
+                this.player1.clickCard(this.p2Base);
+                expect(this.player1).toBeAbleToSelectExactly([this.mandalorianWarrior, this.battlefieldMarine, this.bokatanKryze, this.protectorOfTheThrone, this.allianceXwing, this.jedhaAgitator]);
+                expect(this.player1).toHavePassAbilityButton();
+                // prompt does not change between 2 effects of bo katan ability
+                this.player1.clickCardNonChecking(this.protectorOfTheThrone);
+                expect(this.player1).toBeAbleToSelectExactly([this.mandalorianWarrior, this.battlefieldMarine, this.bokatanKryze, this.protectorOfTheThrone, this.allianceXwing, this.jedhaAgitator]);
+                expect(this.player1).toHaveChooseNoTargetButton();
+                this.player1.clickCard(this.allianceXwing);
+                expect(this.allianceXwing.damage).toBe(1);
+                // 1 damage from previously
+                expect(this.protectorOfTheThrone.damage).toBe(2);
+
+                // 2 triggers as we attack with another mandalorian (2 damage to 1 unit)
+                this.bokatanKryze.exhausted = false;
+                this.player2.passAction();
+                this.player1.clickCard(this.bokatanKryze);
+                this.player1.clickCard(this.p2Base);
+                expect(this.player1).toBeAbleToSelectExactly([this.mandalorianWarrior, this.battlefieldMarine, this.bokatanKryze, this.protectorOfTheThrone, this.allianceXwing, this.jedhaAgitator]);
+                expect(this.player1).toHavePassAbilityButton();
+                // prompt does not change between 2 effects of bo katan ability
+                this.player1.clickCardNonChecking(this.battlefieldMarine);
+                expect(this.player1).toBeAbleToSelectExactly([this.mandalorianWarrior, this.battlefieldMarine, this.bokatanKryze, this.protectorOfTheThrone, this.allianceXwing, this.jedhaAgitator]);
+                expect(this.player1).toHaveChooseNoTargetButton();
+                this.player1.clickCard(this.battlefieldMarine);
+                expect(this.battlefieldMarine.damage).toBe(2);
+
+                // 2 triggers as we attack with another mandalorian (2 damage to 1 unit who die on first damage)
+                this.bokatanKryze.exhausted = false;
+                this.player2.passAction();
+                this.player1.clickCard(this.bokatanKryze);
+                this.player1.clickCard(this.p2Base);
+                expect(this.player1).toBeAbleToSelectExactly([this.mandalorianWarrior, this.battlefieldMarine, this.bokatanKryze, this.protectorOfTheThrone, this.allianceXwing, this.jedhaAgitator]);
+                expect(this.player1).toHavePassAbilityButton();
+                // prompt does not change between 2 effects of bo katan ability
+                this.player1.clickCardNonChecking(this.jedhaAgitator);
```

Comment:
In this situation we should use clickCard() because the prompt does change (Jedha Agitator is removed from the list of options). However, if we change to `simultaneous()` as I suggested above, then the prompt will not change and we should keep using `clickCardNonChecking()`.